src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1230,16 +1230,18 @@ bool LinearScan::buildKillPositionsForNode(GenTree* tree, LsraLocation currentLo
 
         // After a Swift call that might throw returns, we expect the error register to be consumed
         // by a GT_SWIFT_ERROR node. However, we want to ensure the error register won't be trashed
-        // before GT_SWIFT_ERROR can consume it
-        // (for example, the PInvoke epilog comes before the error register store).
-        // To do so, mark the register as "busy until next kill".
-        // This means the register will remain busy from the Swift call until the GT_SWIFT_ERROR kills it.
-
-        RegRecord*   swiftErrorRegRec = getRegisterRecord(REG_SWIFT_ERROR);
-        RefPosition* pos              = swiftErrorRegRec->lastRefPosition;
-        assert(pos != nullptr);
-        assert(pos->refType = RefTypeKill);
-        pos->busyUntilNextKill = true;
+        // before GT_SWIFT_ERROR can consume it.
+        // (For example, the PInvoke epilog comes before the error register store.)
+        // To do so, delay the freeing of the error register until the next node.
+        // This only works if the next node after the call is the GT_SWIFT_ERROR node.
+        // (InsertPInvokeCallEpilog should have moved the GT_SWIFT_ERROR node during lowering.)
+        assert(tree->gtNext != nullptr);
+        assert(tree->gtNext->OperIs(GT_SWIFT_ERROR));
+
+        // We could use RefTypeKill, but RefTypeFixedReg is used less commonly, so the check for delayRegFree
+        // during register allocation should be cheaper in terms of TP.
+        RefPosition* pos  = newRefPosition(REG_SWIFT_ERROR, currentLoc, RefTypeFixedReg, tree, RBM_SWIFT_ERROR);
+        pos->delayRegFree = true;
     }
 #endif // SWIFT_SUPPORT
```

Comment:
It probably also makes more sense to do this at the end of x64/arm64 `BuildCall`.